client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js

```diff
@@ -69,126 +107,265 @@ class PaymentMethodStripe extends Component {
+		this.setState( { createSelected: true } );
+	}
+
+	onSelectConnect = () => {
+		this.setState( { createSelected: false } );
+	}
+
+	possiblyRenderSetupPrompt = () => {
+		const { stripeConnectAccount, translate } = this.props;
+		const { connectedUserID } = stripeConnectAccount;
+
+		// If we don't have the new connect UX enabled yet, keep the legacy prompt instead
+		if ( ! config.isEnabled( 'woocommerce/extension-settings-stripe-connect-flows' ) ) {
+			return (
+				<Notice showDismiss={ false } text={ translate( 'To use Stripe you need to register an account' ) }>
+					<NoticeAction href="https://dashboard.stripe.com/register">{ translate( 'Sign up' ) }</NoticeAction>
+				</Notice>
+			);
+		}
+
+		// Do we have any API keys? If so, do not display connect prompt
+		// unless the user specifically asked for it
+		if ( this.hasKeys() && ! this.state.userRequestedConnectFlow ) {
+			return null;
+		}
+
+		// Do we have a Stripe Connect User ID already? If so, do not display connect prompt
+		if ( connectedUserID ) {
+			return null;
+		}
+
+		// Did the user ask for the key based flow? If so, do not display connect prompt
+		if ( this.state.userRequestedKeyFlow ) {
+			return null;
+		}
+
+		// Did we have keys when we started? If so, do not display the connect prompt
+		// unless the user explicitly requested the connect flow
+		if ( this.state.hadKeysAtStart && ! this.state.userRequestedConnectFlow ) {
+			return null;
+		}
+
+		return (
+			<StripeConnectPrompt
+				isCreateSelected={ this.state.createSelected }
+				onSelectCreate={ this.onSelectCreate }
+				onSelectConnect={ this.onSelectConnect }
+			/>
+		);
+	}
+
+	////////////////////////////////////////////////////////////////////////////
+	// Stripe Connect Account details (if any)
+
+	possiblyRenderConnectAccount = () => {
+		const { stripeConnectAccount } = this.props;
+
+		// No account ID? Bail
+		if ( ! stripeConnectAccount.connectedUserID ) {
+			return null;
+		}
 
 		return (
-			<div>
-					{ this.renderEditKeyTextbox(
-						isLiveMode ? method.settings.secret_key : method.settings.test_secret_key,
-						secretLabel,
-						translate( 'Enter your secret key from your Stripe.com account' )
-					) }
-					{ this.renderEditKeyTextbox(
-						isLiveMode ? method.settings.publishable_key : method.settings.test_publishable_key,
-						publishableLabel,
-						translate( 'Enter your publishable key from your Stripe.com account' )
-					) }
-			</div>
+			<StripeConnectAccount
+				stripeConnectAccount={ stripeConnectAccount }
+			/>
 		);
 	}
 
-	onDone = ( e ) => {
-		if ( hasStripeValidCredentials( this.props.method ) ) {
-			this.props.onDone( e );
+	////////////////////////////////////////////////////////////////////////////
+	// Live vs Test Mode and API Key Fields
+
+	onSelectLive = () => {
+		this.props.onEditField( 'testmode', 'no' );
+	}
+
+	onSelectTest = () => {
+		this.props.onEditField( 'testmode', 'yes' );
+	}
+
+	possiblyRenderModePromptAndKeyFields = () => {
```

Comment:
This function is doing a lot of things and could stand to be broken up as well, I think.

Reply:
I actually did have it as two before :) The visibility logic was identical, so I combined them - I will split them again and pull the shared visibility logic into a helper method

Reply:
Simplified a little in https://github.com/Automattic/wp-calypso/pull/17197/commits/4c6c545a3975ac1f5943dbfbe92b93c35544a937